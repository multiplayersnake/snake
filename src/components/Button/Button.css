:root {
  --button-width: var(--normal-button-width);
  --button-height: var(--normal-button-height);
}

.button {
  min-width: var(--button-width);
  height: var(--button-height);
  padding-left: var(--gap-5);
  padding-right: var(--gap-5);
  color: var(--violet);
  font-family: var(--impact-font);
  font-size: var(--gap-6);
  border-radius: var(--normal-border-radius);
  background-color: var(--yellow);
  position: relative;
  box-shadow: var(--inset-black-shadow);
  cursor: pointer;
}

.button:after {
  position: absolute;
  display: block;
  content: '';
  width: 100%;
  height: 100%;
  top: 0;
  left: 0;
  border-radius: var(--normal-border-radius);
}

.button:hover:after {
  background-color: var(--transparent-blue);
}

.button:focus {
  box-shadow: var(--gray-shadow), var(--inset-black-shadow);
}
