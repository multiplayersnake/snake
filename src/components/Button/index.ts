export { default } from './Button';
export { default as NavButton } from './NavButton';
