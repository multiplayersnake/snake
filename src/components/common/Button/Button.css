:root {
  --button-width: var(--normal-button-width);
  --button-height: var(--normal-button-height);
}

.button {
  color: var(--black);
  font-family: var(--main-font);
  font-size: var(--gap-6);
  font-weight: normal;
  border-radius: var(--gap5-1);
  background: var(--button-background-color);
  position: relative;
  box-shadow: 0 4px 4px rgba(0, 0, 0, 0.5);
  cursor: pointer;
}

.regular-button {
  min-width: var(--button-width);
  height: var(--button-height);
  padding-left: var(--gap-5);
  padding-right: var(--gap-5);
}

.icon-button {
  width: 32px;
  height: 32px;
  display: flex;
  justify-content: center;
  align-items: center;
}

.button:after {
  position: absolute;
  display: block;
  content: '';
  width: 100%;
  height: 100%;
  top: 0;
  left: 0;
  border-radius: var(--normal-border-radius);
}

.button:hover,
.button:focus {
  box-shadow: 0 0 5px rgb(255, 255, 255);
}
