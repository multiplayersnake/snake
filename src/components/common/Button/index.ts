export * from './Button';
export * from './NavButton';
