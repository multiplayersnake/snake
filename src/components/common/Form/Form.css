.form {
  display: flex;
  flex-flow: column;
  align-items: center;
  padding: var(--gap-4);
  gap: var(--gap-5);
  border-radius: var(--normal-border-radius);
}
