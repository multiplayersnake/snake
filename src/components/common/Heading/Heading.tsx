import React, { FC, HTMLAttributes } from 'react';
import cn from 'classnames';

import './Heading.css';

type HeadingTag = 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6';

type HeadingProps = HTMLAttributes<HTMLHeadingElement> & {
  tag?: HeadingTag;
};

export const Heading: FC<HeadingProps> = (props) => {
  const { children, className, tag: Tag = 'h1', ...rest } = props;

  return (
    <Tag {...rest} className={cn('heading', Tag, className)}>
      {children}
    </Tag>
  );
};
