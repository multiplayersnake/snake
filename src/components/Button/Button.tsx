import React, { FC, ButtonHTMLAttributes } from 'react';

import './Button.css';

const Button: FC<ButtonHTMLAttributes<HTMLButtonElement>> = (props) => {
  const { children, ...rest } = props;

  return (
    <button {...rest} className="button">
      {children}
    </button>
  );
};

export default Button;
