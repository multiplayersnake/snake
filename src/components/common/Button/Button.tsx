import React, { FC, ButtonHTMLAttributes, useRef, useCallback } from 'react';
import cn from 'classnames';

import './Button.css';
import mscHover from '../../../assets/sound/button_hover.mp3';

type ButtonProps = ButtonHTMLAttributes<HTMLButtonElement>;

export const Button: FC<ButtonProps> = (props) => {
  const { type = 'button', children, className, ...rest } = props;

  const ref = useRef(null);
  const hover = useCallback(() => {
    const obj = ref.current as HTMLAudioElement;
    obj.volume = 0.2;
    obj
      .play()
      .then(function () {
        // console.log('playing');
      })
      .catch(function (reason) {
        console.log(reason);
      });
  }, []);

  return (
    <button {...rest} className={cn('button', className)} type={type} onMouseEnter={hover}>
      <audio ref={ref} src={mscHover} autoPlay={false} preload={'auto'} />
      {children}
    </button>
  );
};

export default Button;
