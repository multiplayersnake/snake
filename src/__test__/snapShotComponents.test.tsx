import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../components/Button';
import Input from '../components/Input';
import Heading from '../components/Heading';
import LogoutLink from '../components/LogoutLink';
import TextArea from '../components/TextArea';
import Form from '../components/Form';
import Message from '../components/ForumPageComponents/Message';
import Topic from '../components/ForumPageComponents/Topic';
import InfoPanel from '../components/MainPageComponents/InfoPanel';
import MainMenuItem from '../components/MainPageComponents/MainMenuItem';
import NavMenu from '../components/NavMenu';

import { MenuActionType } from '../types/mainMenu';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureStore from '../store/store';

const store = configureStore({});

it('Button', () => {
  const tree = renderer.create(<Button onClick={() => ''}>Test Button</Button>).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Form', () => {
  const tree = renderer.create(<Form onSubmit={() => ''}>Test Form</Form>).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Message', () => {
  const tree = renderer.create(<Message content={''} author={''} dateTime={''} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Topic', () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <Topic id={1} newCount={1} href={''} mesCount={1} content={''} author={''} dateTime={''} />
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Input', () => {
  const tree = renderer.create(<Input label={'Input'} onChange={() => ''} onClick={() => ''} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Heading', () => {
  const tree = renderer.create(<Heading tag={'h1'}>Test Heading</Heading>).toJSON();
  expect(tree).toMatchSnapshot();
});

it('LogoutLink', () => {
  const tree = renderer.create(<LogoutLink onClick={() => ''} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('InfoPanel', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <InfoPanel />
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('MainMenuItem', () => {
  const action: MenuActionType = MenuActionType.Login;
  const item = { title: '', action: action, path: '', authorizedOnly: true };
  const tree = renderer.create(<MainMenuItem authorized={true} item={item} onAction={() => ''} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('NavMenu', () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <NavMenu onAction={() => ''} />
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('TextArea', () => {
  const tree = renderer.create(<TextArea label={'Test TextArea'} />).toJSON();
  expect(tree).toMatchSnapshot();
});
