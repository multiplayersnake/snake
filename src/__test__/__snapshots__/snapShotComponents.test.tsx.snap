// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button 1`] = `
<button
  className="button"
  onClick={[Function]}
  onMouseEnter={[Function]}
  type="button"
>
  <audio
    autoPlay={false}
    preload="auto"
    src={Object {}}
  />
  Test Button
</button>
`;

exports[`Form 1`] = `
<form
  className="form"
  onSubmit={[Function]}
>
  Test Form
</form>
`;

exports[`Heading 1`] = `
<h1
  className="heading h1"
>
  Test Heading
</h1>
`;

exports[`InfoPanel 1`] = `
<div
  className="panel left-panel"
>
  <div
    className="heading h6"
  >
    Позывной:
  </div>
  <div
    className="heading h6"
  />
  <div
    className="flex-wrapper"
  />
  <div>
    Валюта:
  </div>
  <div>
     
    <img
      alt="coin"
      className="img-in-line"
      src={Object {}}
    />
  </div>
  <div
    className="flex-wrapper"
  />
  <div>
    Награды: 
  </div>
  <div>
     
    <img
      alt="award"
      className="img-in-line"
      src={Object {}}
    />
  </div>
</div>
`;

exports[`Input 1`] = `
<label
  className="input-container"
>
  <input
    className="input"
    onChange={[Function]}
    onClick={[Function]}
  />
  <span
    className="input-label"
  >
    Input
  </span>
</label>
`;

exports[`Message 1`] = `
<div
  className="message text-field"
>
  <div
    className="message-author"
  >
    
        
    ??.??.???? ??:??
  </div>
  <div
    className="message-edit"
  >
    <button
      className="imgButton message-button"
      onClick={[Function]}
      type="button"
    >
      <img
        alt="Редактировать"
        height="16px"
        src={Object {}}
        width="16px"
      />
    </button>
  </div>
  <div
    className="message-delete"
  >
    <button
      className="imgButton message-button"
      onClick={[Function]}
      type="button"
    >
      <img
        alt="Удалить"
        height="16px"
        src={Object {}}
        width="16px"
      />
    </button>
  </div>
  <div
    className="message-content"
  >
    <div />
  </div>
</div>
`;

exports[`TextArea 1`] = `
<label
  className="text-area-container"
>
  <span
    className="text-area-label"
  >
    Test TextArea
  </span>
  <textarea
    className="text-area"
  />
</label>
`;

exports[`Topic 1`] = `
<a
  href=""
  onClick={[Function]}
>
  <div
    className="topic text-field"
  >
    <div
      className="topic-dateTime text-field"
    >
      Создана: 
      <b>
        ??.??.???? ??:??
      </b>
    </div>
    <div
      className="topic-author text-field"
    >
      Автор: 
      <b>
        0
      </b>
    </div>
    <div
      className="topic-mes-count text-field"
    >
      Сообщений: 
      1
      <b>
         (1)
      </b>
    </div>
    <div
      className="topic-edit"
    >
      <button
        className="imgButton topic-button"
        onClick={[Function]}
        type="button"
      >
        <img
          alt="Редактировать"
          height="16px"
          src={Object {}}
          width="16px"
        />
      </button>
    </div>
    <div
      className="topic-delete topic_hidden"
    >
      <button
        className="imgButton topic-button"
        onClick={[Function]}
        type="button"
      >
        <img
          alt="Удалить"
          height="16px"
          src={Object {}}
          width="16px"
        />
      </button>
    </div>
    <div
      className="topic-content"
    >
      <div />
    </div>
  </div>
</a>
`;
